frontend/src/app/modules/album/components/create-edit-album/create-edit-album/create-edit-album.component.ts

```diff
@@ -0,0 +1,168 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { AlbumFull } from 'src/app/models/album/album-full';
+import { Song } from 'src/app/models/song/song';
+import { AlbumEdit } from 'src/app/models/album/album-edit';
+import { AlbumRegion } from 'src/app/models/album/album-region';
+import { AuthorType } from 'src/app/models/enums/author-type.enum';
+import { AlbumService } from 'src/app/services/album.service';
+import { Subject } from 'rxjs';
+import { switchMap, takeUntil } from 'rxjs/operators';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-create-edit-album',
+  templateUrl: './create-edit-album.component.html',
+  styleUrls: ['./create-edit-album.component.sass']
+})
+export class CreateEditAlbumComponent implements OnInit, OnDestroy {
+  album: AlbumFull = {
+    id: 0,
+    authorType: AuthorType.artist,
+    description: '',
+    iconURL: '',
+    isPublished: false,
+    isSingle: false,
+    name: 'Album name',
+    region: AlbumRegion.usa,
+    releaseYear: new Date().getFullYear(),
+    artist: undefined,
+    group: undefined,
+    songs: {} as Song[]
+  };
```

Comment:
Code duplicates here. Write a function and use it:
getBasicAlbumFull = () => ({
  id: ...,
})